byron/ledger/executable-spec/src/Ledger/Core.hs

```diff
@@ -374,12 +385,42 @@ instance Relation (Set (a, b)) where
     where
       d1' = toSet d1
 
+  vmax <=◁ r = Set.filter ((<= vmax) . fst) $ r
+
   r ▷<= vmax = Set.filter ((<= vmax) . snd) $ r
 
   r ▷>= vmax = Set.filter ((>= vmax) . snd) $ r
 
   size = fromIntegral . Set.size
 
+instance Relation [(a, b)] where
+  type Domain [(a, b)] = a
+  type Range [(a, b)] = b
+
+  singleton a b = [(a, b)]
+
+  dom = toSet . fmap fst
+
+  range = toSet . fmap snd
+
+  s ◁ r = filter ((`Set.member` toSet s) . fst) r
+
+  s ⋪ r = filter ((`Set.notMember` toSet s) . fst) r
+
+  r ▷ s = filter ((`Set.notMember` toSet s) . snd) r
+
+  (∪) = (++)
+
+  -- In principle a list of pairs allows for duplicated keys.
+  d0 ⨃ d1 = d0 ++ toList d1
```

Comment:
the same is true for `instance Relation (Set (a, b))` (we can have duplicate keys but distinct pairs) - and in that case we do still take care to exclude `dom d1` from `d0` - we could do the same here

Reply:
Hmm, I don't. I'd say having a more liberal instance for `[(a, b)]` is fine here, but maybe because I haven't thought about the properties long enough ;) Do you see a potential problem with this instance?

Reply:
the Relation property tests we have now don't cover Union override, so it won't cause a problem yet. (Once we test union override, this implementation will probably fail some properties we'd consider reasonable, but not a problem now). 

One thing to do now though is to hook your new relation instance into the Relation property tests (`Ledger.Relation.Properties`)